src/network/http/websocket.cpp

```diff
@@ -173,6 +154,14 @@ namespace fc { namespace http {
               return _ws_connection->get_request_header(key);
             }
 
+            virtual std::string get_host()override
+            {
+               auto header = get_request_header("BS-Forwarded");
+               if (header.length() > 2)
+                  return header;
+               return _ws_connection->get_host();
```

Comment:
I think `get_host` is wrong, according to the docs it returns the host component of the requested URI, not the remote host.

Reply:
I will investigate.